packages/11ty/_plugins/figures/index.js

```diff
@@ -0,0 +1,17 @@
+const Figure = require('./figure')
+const iiifConfig = require('./config')
+
+module.exports = function (eleventyConfig, options = {}) {
+  eleventyConfig.addGlobalData('iiifConfig', iiifConfig(eleventyConfig))
+
+  eleventyConfig.on('eleventy.before', async () => {
+    let { figure_list: figureList } = eleventyConfig.globalData.figures
+    figureList = await Promise.all(
+      figureList.map((data) => {
+        const figure = new Figure(eleventyConfig, data)
+        return figure.create()
```

Comment:
Making the statements in the `create` method part of the `Figure.constructor` would be more idiomatic however they may be asynchronous in which case refactoring the the `Figure` class as a FigureFactory class that returns a single method may another approach, for example
```suggestion
        const factory = new FigureFactory(eleventyConfig)
        return factory(data) 
```
the `factory` could also simply be a module with the a `create` method
```
        const factory = new FigureFactory(eleventyConfig)
        return factory.create(data) 
```